src/Core/Scheduler/Providers/Raster_Provider.js

```diff
@@ -73,81 +51,70 @@ export default {
             throw new Error('layer.url is required');
         }
 
-        const extention = getExtension(layer.url).toLowerCase();
-        const format = `vector/${extention}`;
         layer.options = layer.options || {};
+        // KML and GPX specifications all says that they should be in
+        // EPSG:4326. We still support reprojection for them through this
+        // configuration option
+        layer.projection = layer.projection || 'EPSG:4326';
+        const parentCrs = layer.parentLayer.extent.crs();
```

Comment:
An alternative to storing the `parentLayer` would be to pass it as an extra argument `preprocessDataLayer`.
WDYT?

Reply:
I thought about that too. I didn't because I expect we'll also need parentLayer in other parts (`executeCommand` ?), but I might be mistaken. The advantage of your solution is that attaching `parentLayer` to layer could be decided independently for each provider.

That being said, for me, having a reference to the parentLayer on the child layer makes sense. At the very least, it's not something neither unexpected nor "dangerous" in my opinion. That's why I did it like that.

Thoughts?

Reply:
The problem is when you want to add the same layer twice (e.g: you have 2 GlobeView and want to add the same color layer to each).
But I'm not sure other parts of iTowns handle this edge-case correctly, so we may want to disable it entirely (e.g: prevent adding a `Layer` instance to another `Layer`)

Reply:
Oh yes, I assumed this was just not supported. Do you think it'll be interesting to support it? If yes, I'm modifying this.

Reply:
Yeah, I'd prefer not adding the `parentLayer` to `Layer` for now, if the other proposal is ok for you (= adding a param to `preprocessLayer`).